src/lib/utils.ts

```diff
@@ -1,6 +1,40 @@
 import { type ClassValue, clsx } from "clsx";
 import { twMerge } from "tailwind-merge";
+import type { V0ServerJson } from "@/generated/types.gen";
 
 export function cn(...inputs: ClassValue[]) {
   return twMerge(clsx(inputs));
 }
+
+/**
+ * Checks if a server is a Virtual MCP server by examining metadata.
+ * Virtual MCP servers have kubernetes_kind set to "VirtualMCPServer" in their metadata.
+ * transportUrl = URL where the MCP server is accessible (e.g. "https://mcp.example.com/servers/my-server")
+ */
+export function isVirtualMCPServer(server: V0ServerJson): boolean {
```

Comment:
this should fully be handled in BE... I mean, as a temporary measure it's fine, but we should not need this logic on FE